legend-engine-language-pure-dsl-service-execution/src/main/java/org/finos/legend/engine/language/pure/dsl/service/execution/AbstractServicePlanExecutor.java

```diff
@@ -122,19 +136,82 @@ protected AbstractServicePlanExecutor(String servicePath, String executionPlanRe
         this(servicePath, executionPlanResource, DEFAULT_ALLOW_JAVA_COMPILATION, storeExecutors);
     }
 
+    @Override
     public String getServicePath()
     {
         return this.servicePath;
     }
 
+    @Override
     public PlanExecutorInfo getPlanExecutorInfo()
     {
         return this.executor.getPlanExecutorInfo();
     }
 
+    @Override
+    public ResultType getResultType()
+    {
+        if (this.plan instanceof SingleExecutionPlan)
+        {
+            return ((SingleExecutionPlan) this.plan).rootExecutionNode.resultType;
+        }
+        else if(this.plan instanceof CompositeExecutionPlan)
+        {
+            List<ResultType> resultTypes = ((CompositeExecutionPlan) this.plan).executionPlans.values()
+                    .stream()
+                    .map(x -> x.rootExecutionNode.resultType)
+                    .collect(Collectors.toList());
+
+            Assert.assertFalse(resultTypes.isEmpty(), () -> "Composite plan without any actual plan?");
+            Assert.assertTrue(resultTypes.stream().allMatch(x -> x.getClass().equals(resultTypes.get(0).getClass())), () -> "Composite with different result types among plans?");
+            return resultTypes.get(0);
+        }
+        else
+        {
+            throw new UnsupportedOperationException("Not supported: " + this.plan.getClass());
+        }
+    }
+
+    protected ExecutionBuilder executionBuilder(ServiceRunnerInput serviceRunnerInput, StreamProvider streamProvider)
```

Comment:
I think newExecutionBuilder would be a better name, so that it's symmetric with the other factory methods for constructing an ExecutionBuilder.